### PR TITLE
bug: `LocalPool` swallowing logs in `iroh-blobs` tests

### PR DESCRIPTION
**Pull request by ramfox**
## Description

While investigating the flaky CI failure of the `localpool::test_drop` test, I found an example of `LocalPool` eating logging.

The logs inside the `test_drop` print, but any logs written in `LocalPool::spawn_detached` are lost.

run `cargo test --package iroh-blobs test_drop -- --nocapture` to view.

I also had to change from using `tracing_subscriber::fmt::try_init()` to `iroh_test::logging::setup()` in the test to get ANY logging to show.

Related to #2577

## Change checklist

- [ ] Self-review.
- [ ] Documentation updates following the [style guide](https://rust-lang.github.io/rfcs/1574-more-api-documentation-conventions.html#appendix-a-full-conventions-text), if relevant.
- [ ] Tests if relevant.
- [ ] All breaking changes documented.

### CODE REVIEW

File: iroh-blobs/src/util/local_pool.rs [hunk 1]
```diff
@@ -568,6 +572,7 @@ mod tests {
         tokio::time::sleep(Duration::from_millis(100)).await;
         // drop x at the end. we will never get here when the future is
         // no longer polled, but drop should still be called
+        tracing::debug!("Dropping {x:?}");
```
Comment (ramfox) [inline iroh-blobs/src/util/local_pool.rs:575]: @dignifiedquire should this line still be caught in logging? It's not currently.

**Comment by flub** [comment]
Still looking into this.  FWIW the `iroh_test::logging::setup` has never worked with tests using multiple threads, so that's not really any different with the LocalPool.

The problem with using a global subscriber is that all tests use the same subscriber and IIRC your tests output starts flowing into the wrong tests due to how test isolation and output capturing works.

**Comment by flub** [comment]
Oops, I thought this was an issue and it's a PR...  I replaced this in #2735

**Comment by flub** [comment]
The other PR did get merged

**Pull request closed without merging**